test/unit/messaging/messaging.spec.ts

```diff
@@ -1693,6 +1995,319 @@ describe('Messaging', () => {
+              aps: {
+                alert: 'test.alert',
+              },
+            },
+          },
+        },
+      },
+      {
+        label: 'All APNS properties',
+        req: {
+          apns: {
+            headers: {
+              h1: 'v1',
+              h2: 'v2',
+            },
+            payload: {
+              aps: {
+                alert: {
+                  titleLocKey: 'title.loc.key',
+                  titleLocArgs: ['arg1', 'arg2'],
+                  locKey: 'body.loc.key',
+                  locArgs: ['arg1', 'arg2'],
+                  actionLocKey: 'action.loc.key',
+                },
+                badge: 42,
+                sound: 'test.sound',
+                category: 'test.category',
+                contentAvailable: true,
+                threadId: 'thread.id',
+              },
+              customKey1: 'custom.value',
+              customKey2: {nested: 'value'},
+            },
+          },
+        },
+        want: {
+          apns: {
+            headers: {
+              h1: 'v1',
+              h2: 'v2',
+            },
+            payload: {
+              aps: {
+                'alert': {
+                  'title-loc-key': 'title.loc.key',
+                  'title-loc-args': ['arg1', 'arg2'],
+                  'loc-key': 'body.loc.key',
+                  'loc-args': ['arg1', 'arg2'],
+                  'action-loc-key': 'action.loc.key',
+                },
+                'badge': 42,
+                'sound': 'test.sound',
+                'category': 'test.category',
+                'content-available': 1,
+                'thread-id': 'thread.id',
+              },
+              customKey1: 'custom.value',
+              customKey2: {nested: 'value'},
+            },
+          },
+        },
+      },
+      {
+        label: 'APNS contentAvailable explicitly false',
+        req: {
+          apns: {
+            payload: {
+              aps: {
+                contentAvailable: false,
+              },
+            },
+          },
+        },
+        want: {
+          apns: {
+            payload: {
+              aps: {},
+            },
+          },
+        },
+      },
+    ];
+
+    validMessages.forEach((config) => {
+      it(`should serialize well-formed Message: ${config.label}`, () => {
+        // Wait for the initial getToken() call to complete before stubbing https.request.
+        return mockApp.INTERNAL.getToken()
+          .then(() => {
+            httpsRequestStub = sinon.stub(https, 'request');
+            httpsRequestStub.callsArgWith(1, mockResponse).returns(mockRequestStream);
+            const req = config.req;
+            req.token = 'mock-token';
+            return messaging.send(req);
+          })
+          .then(() => {
+            expect(requestWriteSpy).to.have.been.calledOnce;
+            const requestData = JSON.parse(requestWriteSpy.args[0][0]);
+            const want = config.want || config.req;
```

Comment:
Suggestion: maybe call it expectedServerReq or expectedReq or something of the like, instead of want ?

Reply:
Done